File: 346/346.py

def getDecimalVal(numOnes, base):
  retVal = 0

  exponent = numOnes - 1;
  while (exponent >= 0):
    retVal += (base ** exponent)

    exponent = exponent - 1

  return retVal

mxx = 10 ** 12

multMap = {1: True}
numOnes = 3

while getDecimalVal(numOnes, 2) < mxx:
  for base in xrange(2, mxx):
    dVal = getDecimalVal(numOnes, base)
    if dVal > mxx:
      break;

    # print numOnes, base, dVal

    multMap[dVal] = True
    
  numOnes = numOnes + 1

# multMap[dVal] = True
# print multMap
ssum = 0
for key in multMap:
  ssum = ssum + key
print ssum

# print getDecimalVal(30, 2)

File: 346/346.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int getDecimalVal(int numOnes, int base) {
  int retVal = 0;

  int exponent = numOnes - 1;
  while (exponent >= 0) {
    retVal += pow(base, exponent);

    exponent --;
  }

  return retVal;
}


int main() {
  int numOnes = 1;

  // while (numOnes < 4) {
  //   for (int base = 2; base < 8; base ++) {
  //     printf("%i\n", getDecimalVal(numOnes, base));
  //   }

  //   numOnes ++;
  // }

  printf("%i\n", getDecimalVal(2, 6)); 
  
  return 0;
}

